tests/testthat/test-query.R

```diff
@@ -0,0 +1,34 @@
+test_that("query strings are properly parsed", {
+  expect_equal(parse_query("?a=1"), list(a = "1"))
+  expect_equal(parse_query("b=2"), list(b = "2"))
+  expect_equal(parse_query("a=1&b=2&c=url%20encoded"),
+               list(a = "1", b = "2", c = "url encoded"))
+})
+
+
+test_that("special characters in query strings are handled properly", {
+  expect_equal(parse_query("?a=1+.#"), list(a = "1 .#"))
+  expect_equal(parse_query("?a=a%20b"), list(a = "a b"))
+  expect_equal(parse_query("?a = %2C%2B%2F%3F%25%26"), list(a = ",+/?%&"))
+})
+
+
+test_that("null an empty strings return empty list", {
+  expect_equal(parse_query(NULL), list())
+  expect_equal(parse_query(""), list())
+  expect_equal(parse_query("?"), list())
+})
+
+
+test_that("incomplete query strings are errors", {
+  expect_error(parse_query("a="), "Incomplete query for 'a'")
+  expect_error(parse_query("a=1&b=&c&d=1"), "Incomplete query for 'b', 'c'")
```

Comment:
what about query strings without values at all (so `?a` rather than `?a=`)? these are relative commonly used

Reply:
Plumber would have previously dropped these, so we never used to support them. We could add support for them later I guess, along with changes to `porcelain::porcelain_input_query` to add an empty type? 

How is this interpreted generally? I've not come across it? Is it essentially `?a=true`?

Reply:
Yeah, sure. You mean how is it parsed? Think it just varies a lot by framework. Looks like in asp.net `a` would actually get treated as a *value* where the key is null! If you mean what's the intended usage, yeah a boolean flag (I've most commonly seen it as a feature switch, like GET http://whatever?useNewFeature)

Reply:
I guess just worth having a test to show what the current behaviour is here